python/plugins/processing/algs/gdal/rasterize_over.py

```diff
@@ -21,38 +21,59 @@
 __date__ = 'September 2013'
 __copyright__ = '(C) 2013, Alexander Bruy'
 
-import os
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
 
-from qgis.core import QgsProcessingUtils
+import os
 
 from qgis.PyQt.QtGui import QIcon
 
+from qgis.core import (QgsRasterFileWriter,
+                       QgsProcessingParameterDefinition,
+                       QgsProcessingParameterFeatureSource,
+                       QgsProcessingParameterField,
+                       QgsProcessingParameterRasterLayer,
+                       QgsProcessingParameterNumber,
+                       QgsProcessingParameterString,
+                       QgsProcessingParameterEnum,
+                       QgsProcessingParameterExtent,
+                       QgsProcessingParameterBoolean,
+                       QgsProcessingParameterRasterDestination)
 from processing.algs.gdal.GdalAlgorithm import GdalAlgorithm
 from processing.algs.gdal.GdalUtils import GdalUtils
 
-from processing.tools import dataobjects
-
 pluginPath = os.path.split(os.path.split(os.path.dirname(__file__))[0])[0]
 
 
 class rasterize_over(GdalAlgorithm):
 
     INPUT = 'INPUT'
-    INPUT_RASTER = 'INPUT_RASTER'
     FIELD = 'FIELD'
-
-    def icon(self):
-        return QIcon(os.path.join(pluginPath, 'images', 'gdaltools', 'rasterize.png'))
+    INPUT_RASTER = 'INPUT_RASTER'
+    ADD = 'ADD'
 
     def __init__(self):
         super().__init__()
 
     def initAlgorithm(self, config=None):
-        self.addParameter(ParameterVector(self.INPUT, self.tr('Input layer')))
-        self.addParameter(ParameterTableField(self.FIELD,
-                                              self.tr('Attribute field'), self.INPUT))
-        self.addParameter(ParameterRaster(self.INPUT_RASTER,
-                                          self.tr('Existing raster layer'), False))
+        self.units = [self.tr("Pixels"),
+                      self.tr("Georeferenced units")]
+
+        self.addParameter(QgsProcessingParameterFeatureSource(self.INPUT,
+                                                              self.tr('Input vector layer')))
+        self.addParameter(QgsProcessingParameterRasterLayer(self.INPUT_RASTER, self.tr('Input raster layer')))
+        self.addParameter(QgsProcessingParameterField(self.FIELD,
+                                                      self.tr('Field to use for a burn-in value'),
+                                                      None,
+                                                      self.INPUT,
+                                                      QgsProcessingParameterField.Numeric,
+                                                      optional=False))
+        add_param = QgsProcessingParameterBoolean(self.ADD,
+                                                     self.tr('Adds the new value to the existing raster'),
```

Comment:
"Add burn in values to existing raster values"

Reply:
changed.